src/wallet/use-cases/find-all/find.all.spec.ts

```diff
@@ -0,0 +1,68 @@
+import { Test } from '@nestjs/testing';
+
+import { FindAllWalletsUseCase } from '@/wallet/use-cases';
+import { IWalletRepository } from '@/wallet/interfaces';
+import { AccountType } from '@/shared/constants';
+
+describe('Find All Wallets', () => {
+  let findAll: FindAllWalletsUseCase;
+  let walletRepository: jest.Mocked<IWalletRepository>;
+
+  beforeAll(async () => {
+    const module = await Test.createTestingModule({
+      providers: [
+        FindAllWalletsUseCase,
+        { provide: IWalletRepository, useValue: { findAll: jest.fn() } },
+      ],
+    }).compile();
+    walletRepository = module.get(IWalletRepository);
+    findAll = module.get(FindAllWalletsUseCase);
+  });
+
+  it('should be defined', () => {
+    expect(walletRepository).toBeDefined();
+    expect(findAll).toBeDefined();
+  });
+
+  it('should be able to return all wallets', async () => {
+    const wallets = [
+      {
+        id: '01',
+        account_type: AccountType.CHECKING_ACCOUNT,
+        description: 'Primeira Descrição de carteira.',
+        created_at: new Date(),
+        updated_at: new Date(),
+        deleted_at: null,
+        bank: null,
+        user: null,
+        transactions: null,
+      },
+      {
+        id: '02',
+        account_type: AccountType.CASH,
+        description: 'Segunda Descrição de carteira.',
+        created_at: new Date(),
+        updated_at: new Date(),
+        deleted_at: null,
+        bank: null,
+        user: null,
+        transactions: null,
+      },
+    ];
+
+    walletRepository.findAll.mockResolvedValue(wallets);
+
+    const result = await findAll.execute();
+
+    expect(result.wallets).toEqual(wallets);
+    expect(walletRepository.findAll).toHaveBeenCalledTimes(1);
+  });
+
+  it('should not be able to return wallets if they have not been found', async () => {
```

Comment:
Acho que eu mudaria o texto, pq vc diz que ela não deve ser possível retornar, oq parece que ela deve retornar um erro. Algo como "Should return empty array if no wallet is found"

Reply:
Mudei aqui Levi, por estarmos seguindo um padrão de "should be able" e "should not be able", eu mantive o "should not", mas encurtei o tamanho do texto como você disse.